bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerWorkAreaHandler.java

```diff
@@ -134,19 +131,22 @@ public void initialize() {
             logger.warn("No AutomowerBridgeHandler found for thingId {}", this.thingId);
         }
 
+        updateStatus(ThingStatus.UNKNOWN); // Set to UNKNOWN initially
+
+        scheduler.execute(() -> completeInitAsync());
+    }
+
+    private void completeInitAsync() {
         // Initial poll to create and set the status of the channels
         poll();
-
         updateStatus(ThingStatus.ONLINE);
```

Comment:
The poll(); can set the thing status to offline, but this statusupdate to online immediate overwrites it. 

There are multiple ways to work around it, but maybe returning a simple boolean from poll() would indicate succes or not. And set the state accordingly.

Reply:
Moved `updateStatus(ThingStatus.ONLINE);` to `AutomowerWorkAreaHandler.updateChannels()`.
Same logic as in `AutomowerWorkAreaHandler`.